### ISSUE TITLE
S03.01 -> TODO 46 -> Misleading TODO

### ISSUE THREAD

**Issue by rsd-raul** [udacity/ud851-Sunshine#19]
Not sure if it's intended but the TODO number 46 in MainActivity reads:

`// TODO (46) Instead of setting the text to "", set the adapter to null before refreshing`

And I think it should refer to the adapter's data instead of the adapter itself.

As something like `mForecastAdapter.setWeatherData(null);` I'm assuming is the answer.

**Comment by mnemonicflow** [comment]
@rsd-raul Yes, that's the answer! Setting the adapter itself to null would result in a NullPointerException

**Comment by artem-p** [comment]
Agree.

**Comment by imonhf** [comment]
Agree. raul is right.

**Comment by funnygerman** [comment]
@JeremySilverTongue the issue is still there

**Comment by JeremySilverTongue** [comment]
Hey, Tim. Unfortunately I'm no longer working at Udacity, so can't fix
this. @jwill is now in charge of all our Android course maintenance.

**Comment by StefanSamson** [comment]
Agree.

**Comment by phillies** [comment]
I just ran into this problem that following the TODO you break the app but with a little rewording this would be fine. I looked here and found this is an open issue for almost 2 years. Why does no one approve the pull request (https://github.com/udacity/ud851-Sunshine/pull/73) for this?!?

**Comment by JeremySilverTongue** [comment]
I reached out to people I still know at Udacity and it seems no one at the company is responsible for this code anymore... Sorry. Wish it weren't so.
